File: reverse-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseString = function(s) {
    var reversed = '';
    var i;

    for (i = s.length - 1; i >= 0; i--) {
        reversed += s[i];
    }

    return reversed;
};

var main = function() {
  console.log(reverseString('hello'));
}

if (require.main === module) {
  main();
}

File: reverse-string/reverse-string.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        i = 0
        j = len(s) - 1

        l = list(s)

        while i < j:
            l[i], l[j] = l[j], l[i]
            i += 1
            j -= 1

        return ''.join(l)

if __name__ == '__main__':
    s = Solution()
    print(s.reverseString('hello'))
